lib/services/elasticsearch.js

```diff
@@ -40,6 +40,7 @@ const childMappingProperties = ['type'];
 
 // Used for collection emulation
 const
+  HIDDEN_COLLECTION = '_kuzzle_keep',
```

Comment:
Since that hidden collection is an inner Kuzzle mechanism unknown to clients, they do not necessarily know that it exists and that rights must be configured to protect it. So, we must add our own protection ourselves:

* The following methods should reject with a `services.storage.unknown_collection` error if the target collection is HIDDEN_COLLECTION: `search`, `get`, `mGet`, `count`, `create`, `createOrReplace`, `update`, `replace`, `delete`, `deleteByQuery`, `getMapping`, `updateMapping`, `truncateCollection`, `import`, `refreshCollection`, `exists`, `mCreate`, `mCreateOrReplace`, `mUpdate`, `mReplace`, `mDelete`
* `createCollection` should reject with a new `services.storage.collection_reserved` error (see the already existing `services.storage.index_reserved` one in #1481 )
* `collectionExists` must return `false` if the HIDDEN_COLLECTION is checked